src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -391,15 +406,21 @@ private function createChoiceList(array $options)
     {
         if (null !== $options['choice_loader']) {
             return $this->choiceListFactory->createListFromLoader(
-                $options['choice_loader'],
+                null === $options['choice_filter'] ? $options['choice_loader'] : new FilterChoiceLoader($options['choice_loader'], $options['choice_filter']),
                 $options['choice_value']
             );
         }
 
         // Harden against NULL values (like in EntityType and ModelType)
         $choices = null !== $options['choices'] ? $options['choices'] : array();
+        $loader = new CallbackChoiceLoader(function () use ($choices) {
```

Comment:
This adds unnecessary overhead on every existing choice type in applications and prevents the caching factory decorator from reusing the same array of choices anymore since we always get a new hash for a new instance of loader, which is a big regression regarding performances.

Whereas we should be careful that all filtered lists can reuse the same cached choice list if possible in the first place (where I stopped last time IIRC).

Reply:
Hm i see, i didnt realize the hash is based on instans (spl_object_hash) for loaders.

What about storing these loaders in memory here, using `CachingFactoryDecorator::generateHash`. Or only normalize the array choices to a loader in case a filter is given.